app/models/campaign_gift_option.rb

```diff
@@ -78,15 +75,19 @@ def to_builder(*expand)
 
       json.deleted !persisted?
 
-      json.gift_option_amount gift_option_amount do |desc|
+      json.gift_option_amount gift_option_amounts do |desc|
         json.amount do 
-          json.currency desc[:amount][:currency]
-          json.cents desc[:amount][:cents]
+          json.currency desc.amount.currency
+          json.cents desc.amount.cents
+        end
+        if desc.recurrence
```

Comment:
Is there any chance that this comes as an empty string instead of `nil`?

Reply:
Hmmm, I'm not sure I understand what you're asking.

Reply:
If `desc.recurrence` comes as an empty string in that verification, that condition is going to be `true` and something will break in the next lines. I like to use `.present?` to prevent that, but maybe that's not the case.

Reply:
I suspect that wouldn't happen BUT there's no harm in adding `.present?` so I will!